packages/firestore/test/unit/local/web_storage_shared_client_state.test.ts

```diff
@@ -108,6 +128,29 @@ class TestSharedClientSyncer implements SharedClientStateSyncer {
   async getActiveClients(): Promise<ClientId[]> {
     return this.activeClients;
   }
+
+  async applyTargetState(
+    targetId: TargetId,
+    state: QueryTargetState,
+    error?: FirestoreError
+  ): Promise<void> {
+    switch (state) {
+      case 'pending':
+        this.pendingTargets.push(targetId);
+        return;
+      case 'active':
+        this.activeTargets.push(targetId);
+        return;
+      case 'inactive':
+        this.inactiveTargets.push(targetId);
+        return;
+      case 'rejected':
+        this.rejectedTargets[targetId] = error;
```

Comment:
FWIW I'd kind of expect that if a target is in rejectedTargets then it's not in pendingTargets or activeTargets, etc.  But I don't think that's how this works...

You could rename this "targetRejections" or something, but I think you have the same problem with e.g. pendingTargets and activeTargets and I'm not coming up with a good naming solution there. :-/

Your tests don't actually run into this since they just do a single applyTargetState() call which isn't super realistic.

I wonder if it'd make sense to just store a TargetID=>state mapping, rather than have  a per-state list?  Or if you intend to test the sequence of state changes, store a TargetId=>list<state> mapping...  WDYT?

Reply:
The tests merely verify that the states that are correctly written to Local Storage. They don't verify the state transitions. I don't rely on any order of transitions in the code and it will be hard to enforce them across multiple tabs.

That being said, I changed `TestSharedClientSyncer` to contain a mapping from target ID to state, but kept the assertions the same. The mapping gets translated into disjunct arrays in `get sharedClientState()`